iModelCore/ECPresentation/Source/Hierarchies/DataSourceInfo.h

```diff
@@ -111,24 +111,23 @@ struct DataSourceIdentifier
 private:
     BeGuid m_id;
     BeGuid m_hierarchyLevelId;
-    Utf8String m_instanceFilter;
+    std::shared_ptr<InstanceFilterDefinition const> m_instanceFilter;
     bvector<uint64_t> m_index;
 public:
     DataSourceIdentifier() : m_id(), m_hierarchyLevelId() {}
-    DataSourceIdentifier(BeGuid hierarchyLevelId, bvector<uint64_t> index, Utf8String instanceFilter)
+    DataSourceIdentifier(BeGuid hierarchyLevelId, bvector<uint64_t> index, std::shared_ptr<InstanceFilterDefinition const> instanceFilter)
         : m_id(), m_hierarchyLevelId(hierarchyLevelId), m_index(index), m_instanceFilter(instanceFilter)
         {}
-    DataSourceIdentifier(BeGuid id, BeGuid hierarchyLevelId, bvector<uint64_t> index, Utf8String instanceFilter)
+    DataSourceIdentifier(BeGuid id, BeGuid hierarchyLevelId, bvector<uint64_t> index, std::shared_ptr<InstanceFilterDefinition const> instanceFilter)
         : m_id(id), m_hierarchyLevelId(hierarchyLevelId), m_index(index), m_instanceFilter(instanceFilter)
         {}
-    bool operator<(DataSourceIdentifier const& other) const
-        {
-        NUMERIC_LESS_COMPARE(m_id, other.m_id);
-        NUMERIC_LESS_COMPARE(m_hierarchyLevelId, other.m_hierarchyLevelId);
-        VECTOR_LESS_COMPARE(m_index, other.m_index);
-        STR_LESS_COMPARE(m_instanceFilter.c_str(), other.m_instanceFilter.c_str());
-        return false;
-        }
+    // bool operator<(DataSourceIdentifier const& other) const

```

Comment:
I guess this is not needed and can be removed?

Reply:
Yes... Fixed.